kbfsgit/git-remote-keybase/main.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+// Git remote helper for the Keybase file system.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/kbfs/env"
+	"github.com/keybase/kbfs/kbfsgit"
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+)
+
+var version = flag.Bool("version", false, "Print version")
+
+const usageFormatStr = `Usage:
+  git-remote-keybase -version
+
+To run against remote KBFS servers:
+  git-remote-keybase %s <remote> [keybase://<repo>]
+
+To run in a local testing environment:
+  git-remote-keybase %s <remote> [keybase://<repo>]
+
+Defaults:
+%s
+`
+
+func getUsageString(ctx libkbfs.Context) string {
+	remoteUsageStr := libkbfs.GetRemoteUsageString()
+	localUsageStr := libkbfs.GetLocalUsageString()
+	defaultUsageStr := libkbfs.GetDefaultsUsageString(ctx)
+	return fmt.Sprintf(
+		usageFormatStr, remoteUsageStr, localUsageStr, defaultUsageStr)
+}
+
+func start() *libfs.Error {
+	kbCtx := env.NewContext()
+
+	// TODO: Also remove all kbfsgit directories older than 24h.
+	storageRoot, err := ioutil.TempDir(kbCtx.GetDataDir(), "kbfsgit")
+	if err != nil {
+		return libfs.InitError(err.Error())
+	}
```

Comment:
I currently have that `defer` in `start.go`, but yeah it probably makes more sense here.  I'll move it.